jbpm-test-coverage/src/test/java/org/jbpm/test/functional/workitem/RetrySyncWorkItemByJPATest.java

```diff
@@ -15,40 +15,60 @@
  */
 package org.jbpm.test.functional.workitem;
 
+import java.util.Arrays;
 import java.util.Collection;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
 import org.drools.core.process.instance.WorkItemManager;
+import org.drools.persistence.api.PersistenceContext;
+import org.drools.persistence.jpa.JpaPersistenceContext;
+import org.jbpm.persistence.JpaProcessPersistenceContextManager;
 import org.jbpm.test.JbpmTestCase;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.kie.api.runtime.Environment;
+import org.kie.api.runtime.EnvironmentName;
 import org.kie.api.runtime.KieSession;
 import org.kie.api.runtime.manager.RuntimeEngine;
 import org.kie.api.runtime.process.NodeInstance;
 import org.kie.api.runtime.process.ProcessInstance;
 import org.kie.api.task.TaskService;
 import org.kie.api.task.model.TaskSummary;
-import org.kie.internal.runtime.manager.context.EmptyContext;
 
 import static org.junit.Assert.assertEquals;
 
+@RunWith(Parameterized.class)
 public class RetrySyncWorkItemByJPATest extends JbpmTestCase {
 
     private static final String RETRY_WORKITEM_JPA_PROCESS_ID = "org.jbpm.test.retryWorkitem-jpa";
 
-    public RetrySyncWorkItemByJPATest() {
+    private boolean pessimistic;
+
+    public RetrySyncWorkItemByJPATest(boolean pessimistic) {
         super( true, true );
+        this.pessimistic = pessimistic;
+    }
+
+    @Parameterized.Parameters(name = "pessimistic locking={0}")
+    public static Collection<Object[]> pessimisticLocking() {
+        Object[][] data = new Object[][] { { false }, { true } };
+        return Arrays.asList(data);
     }
 
     @Test
     public void workItemRecoveryTestByJPA() {
-        manager = createRuntimeManager( "org/jbpm/test/functional/workitem/retry-workitem-jpa.bpmn2" );
-        RuntimeEngine runtimeEngine = getRuntimeEngine( EmptyContext.get() );
+        addEnvironmentEntry(EnvironmentName.USE_PESSIMISTIC_LOCKING, pessimistic);
+        addWorkItemHandler("ExceptionWorkitem", new ExceptionWorkItemHandler());
+        createRuntimeManager( "org/jbpm/test/functional/workitem/retry-workitem-jpa.bpmn2" );
+        RuntimeEngine runtimeEngine = getRuntimeEngine();
+
         KieSession kieSession = runtimeEngine.getKieSession();
-        WorkItemManager workItemManager = (org.drools.core.process.instance.WorkItemManager) kieSession.getWorkItemManager();
+        Environment env = kieSession.getEnvironment();
+        kieSession.getEnvironment().set(EnvironmentName.PERSISTENCE_CONTEXT_MANAGER, new MyJpaProcessPersistenceContextManager(env));
 
-        workItemManager.registerWorkItemHandler( "ExceptionWorkitem", new ExceptionWorkItemHandler() );
```

Comment:
@afalhambra I think the test is failing because this registration was accidetanlly removed

Reply:
no... actually, it is failing because the branch naming is not the same (afalhambra:JBPM-10117 vs fjtirado:JBPM-10117), therefore even if I put that depends on that PR, CI won't be taking that into account, and this PR will keep on failing until the other gets merged.

The workitem registration happens here now: https://github.com/kiegroup/jbpm/pull/2215/files#diff-4cb4abff6d6d9ad857f4e4a9180b9db59febede9184b4fa394f3c48c73d7c209R64

Reply:
so basically, we need your PR merged first